iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_SenderlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER; ++i)
+    {
+        m_senderRuntime->getMiddlewareSender(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReturnValue)
+{
+    auto receiverPort = m_receiverRuntime->getMiddlewareReceiver(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), receiverPort->m_serviceDescription);
+    EXPECT_EQ(22u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_DefaultArgs)
+{
+    auto receiverPort = m_receiverRuntime->getMiddlewareReceiver(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReceiverlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER + 1; ++i)
+    {
+        m_senderRuntime->getMiddlewareReceiver(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_ReturnValue)
+{
+    auto runnableData = m_senderRuntime->getServiceRegistryChangeCounter();
+
+    // Roudi internally  calls 5 servieces before application registers its service
+    EXPECT_EQ(5u, *runnableData);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_OfferStopOfferService)
+{
+    auto initial_value = m_senderRuntime->getServiceRegistryChangeCounter();
+    EXPECT_EQ(5u, *initial_value);
+
+    m_senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    auto runnableData = m_senderRuntime->getServiceRegistryChangeCounter();
+
+    EXPECT_EQ(6u, *runnableData);
+
+    m_senderRuntime->stopOfferService({"service1", "instance1"});
+    this->InterOpWait();
+
+    EXPECT_EQ(7u, *runnableData);
+}
+
+
+TEST_F(PoshRuntime_test, CreateRunnable_ReturnValue)
+{
+    const uint32_t runnableDeviceIdentifier = 1u;
+    iox::runtime::RunnableProperty runnableProperty(iox::cxx::CString100("testRunnable"), runnableDeviceIdentifier);
+
+    auto runableData = m_senderRuntime->createRunnable(runnableProperty);
+
+    EXPECT_EQ(std::string("/sender"), runableData->m_process);
+    EXPECT_EQ(iox::cxx::CString100("testRunnable"), runableData->m_runnable);
+
+    /// @todo I am passing runnableDeviceIdentifier as 1, but it returns 0, is this expected?
+    // EXPECT_EQ(runnableDeviceIdentifier, runableData->m_runnableDeviceIdentifier);
```

Comment:
Probably not intended but currently unused, need to be kept in mind and revisited when we run on multiple devices.